### ISSUE TITLE
Return of "Docker Cloud Build Status" parsing failure

### ISSUE THREAD

**Issue by fmigneault** [badges/shields#10247]
### Are you experiencing an issue with...

shields.io

### 🐞 Description

It was fixed a long while ago (#6235), but seems to have resurfaced recently.

Example: `https://img.shields.io/docker/cloud/build/Ouranosinc/Magpie` 

![Docker Cloud Build Status](https://img.shields.io/docker/cloud/build/Ouranosinc/Magpie)

### 🔗 Link to the badge

https://img.shields.io/docker/cloud/build/Ouranosinc/Magpie

**Comment by PyvesB** [comment]
Hello @fmigneault 👋🏻 

Thanks for reporting the issue! I think this is the same that I observed in #10241. We essentially need to get started on #9563. Any help would be welcome!

**Comment by fmigneault** [comment]
Hi @PyvesB. 
I have noticed this recent #10241 PR, and started getting failures around the same time. 
Is it directly caused by its changes (known cause why parsing fails)? 

I'm all for unifying the Docker badges as proposed by #9563, but does that mean badges will stop working until this gets implemented? Is the integration planed in a near future? I'm worried about the current state since the issue has been open for over half a year without progress.

**Comment by PyvesB** [comment]
> I have noticed this recent https://github.com/badges/shields/pull/10241 PR, and started getting failures around the same time.
> Is it directly caused by its changes (known cause why parsing fails)?

I suggest reading through #10261.

> Is the integration planed in a near future? I'm worried about the current state since the issue has been open for over half a year without progress.

As said in my previous message, any help is welcome. Shields.io is an open-source project, many changes are driven by community contributions, with only one to two active maintainers who do their best at keeping things together with the limited time they can volunteer to this project. 😉

**Comment by PyvesB** [comment]
The DockerCloud badges have been broken for 1.5 years at this point, with failing tests during that whole period. My understanding is that if we landed #9563:
* We'd be able to get badges that were working in early 2024 back up and running.
* But the impact would be fairly limited. Normal Docker badges, which are still working, have totalled ~200k-300k requests in the past 24 hours, whereas the broken Docker Cloud ones have totalled only ~10k.
* And we'd be incurring technical debt, needing to implement new redirects underneath the hood and increased code complexity.

Given no one has manifested any interest on working on this, I'd like to go ahead and deprecate the broken Docker Cloud badges. @jNullj any strong opinions before I go ahead and do so?

**Comment by fmigneault** [comment]
What would be the alternative if deprecated?

It is not that I have no interest or desire to contribute, but I have no idea what/how to fix the problem as I do not know the entire chain of operations involved. From my point of view, I have been waiting for the eventual merge of https://github.com/badges/shields/issues/9563 to fix the issue.

I can obtain the docker details, but not its build information, even though the builds are there and functional (see image below). 

Is there something else that can be done to get the same result as the mentioned working ~200k-300k requests?

`https://img.shields.io/docker/automated/pavics/magpie`
![Docker Automated build](https://img.shields.io/docker/automated/pavics/magpie)
`https://img.shields.io/docker/cloud/automated/pavics/magpie`
![Docker Cloud Automated build](https://img.shields.io/docker/cloud/automated/pavics/magpie)
`https://img.shields.io/docker/cloud/build/pavics/magpie`
![Docker Cloud Build Status](https://img.shields.io/docker/cloud/build/pavics/magpie)
`https://img.shields.io/docker/image-size/pavics/magpie` (matches the last build size!)
![Docker Image Size](https://img.shields.io/docker/image-size/pavics/magpie) 

<img height="500" alt="Image" src="https://github.com/user-attachments/assets/b668c7cb-ec4a-4d8b-8944-63eac0d04717" />

**Comment by PyvesB** [comment]
Did a bit more archeology, here's my rough understanding of the history behind the build & automated badges, ignoring all other Docker badges for simplicity:
1. Initially there were only DockerHub automated and build badges.
2. DockerCloud automated and build badges were added.
3. DockerCloud and DockerHub were merged irrespective of Shields.io.
4. At that point, DockerHub build badges still worked for old repos created before the merger, whereas new repos needed to use the DockerCloud build ones. However, as DockerHub and DockerCloud had merged, from a user perspective our separation did not make sense and was confusing. Folks would often default to trying out the DockerHub badge but get no results (see #4663 and duplicates referenced from there).
5. DockerHub build badges broke and were sunset (#6529), dropping support for old pre-merger repos. On the bright side, new repos had a less confusing experience, they could just leverage the only remaining build badge available, the DockerCloud one.
6. #9563 was created to consolidate the DockerHub and DockerCloud automated badges into a single Docker automated badge, and make the DockerCloud build badge the one and only Docker build badge (i.e. removing the mention of cloud).
7. DockerCloud automated and build badges broke. Users are left with a working DockerHub automated badge, which as far as I can tell works for both pre-and-post-merger repos (needs double-checking, though).

I don't think there's much value in trying to keep backwards compatibility for the `https://img.shields.io/docker/cloud/automated/...` and `https://img.shields.io/docker/cloud/build/...` routes. We would have typically introduced a redirector, trading a bit of code complexity for keeping old badge URLs up and running. But in this case, the trade-off is not worth it IMHO, as the badges have limited usage and have been broken for such a long time, as highlighted by my previous message.

As a proposed plan of attack:
1. We deprecate the DockerCloud automated and build badges. I'm happy to care of this, as it doesn't require Docker-specific knowledge (which I have little of) and it aligns well with my ongoing effort to clean up the project and make it generally easier to maintain. 
2. We re-introduce a new Docker build badge, using the old route that was deleted in #6529. This is where I'd appreciate a contribution from someone who has more Docker experience and interest. @fmigneault if you wanted to give it a go, our [tutorial](https://github.com/badges/shields/blob/master/doc/TUTORIAL.md) for implementing new badges is pretty good, hopefully allowing for a smooth learning curve. I will dedicate time for the review.

**Comment by fmigneault** [comment]
I agree that there is no value to keep backward compatibility of the cloud automated/build distinction since they are broken anyway. 

I don't think there is an easy fix for the docker build status.
The only endpoint I managed to find where a status success/failed is indicated looks like the following:
`https://hub.docker.com/api/build/v1/pavics/source/?image=pavics%2Fmagpie`

However, that requires to be authenticated with access to the build details to get a valid response.

Unless DockerHub provides a new utility endpoint or a property within `https://registry.hub.docker.com/v2/repositories/pavics/magpie`, I do not think there is a public way to retrieve the information.

**Comment by PyvesB** [comment]
Thanks for the information. Just to make sure I understand correctly, build details are not public? I.e. if you perform an authenticated API request, you'll only have access to data of repos you own?

**Comment by fmigneault** [comment]
Yes, or at least to my understanding, that seems to be the case.

**Comment by PyvesB** [comment]
Thanks for confirming. Could you please raise a feature request for such an API here: https://github.com/docker/roadmap/blob/master/CONTRIBUTING.MD#feature-requests

I'll go ahead and deprecate the DockerCloud badges in the meantime.

**Comment by fmigneault** [comment]
Issue opened: https://github.com/docker/roadmap/issues/863

**Comment by PyvesB** [comment]
Thanks, I've subscribed to the issue. Let's see what comes out of it!
